src/app/fdctl/run/tiles/fd_dedup.c

```diff
@@ -114,13 +122,67 @@ after_frag( void *             _ctx,
             ulong *            opt_tsorig,
             int   *            opt_filter,
             fd_mux_context_t * mux ) {
-  (void)in_idx;
   (void)seq;
   (void)opt_tsorig;
   (void)mux;
 
   fd_dedup_ctx_t * ctx = (fd_dedup_ctx_t *)_ctx;
 
+  if ( FD_UNLIKELY( in_idx < ctx->num_in_idx_to_parse ) ) {
+    /* Transactions coming in from the gossip link are not parsed.
+
+       We'll need to parse it so it's ready for downstream consumers.
+       Equally importantly, we need to parse to extract the signature
+       for dedup.  Just parse it right into the output dcache. */
+
+    /* Increment on GOSSIP_IN_IDX but not VOTER_IN_IDX */
+    FD_MCNT_INC( DEDUP, GOSSIPED_VOTES_RECEIVED, 1UL - in_idx );
+
+    ulong payload_sz = *opt_sz;
+    ulong txn_off = fd_ulong_align_up( payload_sz, 2UL );
+
+    /* Ensure sufficient trailing space for parsing. */
+    if( FD_UNLIKELY( txn_off>FD_TPU_DCACHE_MTU - FD_TXN_MAX_SZ - sizeof(ushort)) ) {
+      FD_LOG_ERR(( "got malformed txn (sz %lu) insufficient space left in dcache for fd_txn_t", payload_sz ));
+    }
+
+    uchar    * dcache_entry = fd_chunk_to_laddr( ctx->out_mem, ctx->out_chunk );
+    fd_txn_t * txn = (fd_txn_t *)(dcache_entry + txn_off);
+    ulong txn_t_sz = fd_txn_parse( dcache_entry, payload_sz, txn, NULL );
+    if( FD_UNLIKELY( !txn_t_sz ) ) {
+      FD_LOG_WARNING(( "fd_txn_parse failed for gossiped vote" ));
```

Comment:
I'd call this an error condition, other tiles should not be sending invalid txns here (they must have sigverified)